From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jeremy Apthorp <jeremya@chromium.org>
Date: Thu, 18 Oct 2018 17:06:56 -0700
Subject: expose-net-observer-api.patch

Expose URLRequestContextGetter::{Add,Remove}Observer.
This patch should be removed once we switch to using the net service (cc @robo)

diff --git a/net/url_request/url_request_context_getter.h b/net/url_request/url_request_context_getter.h
index def76d3902f30ae1384ae5f629fa116b3975fa86..e5ee0bfa93f497f18bdb3ad7cd51a6726dc87f3e 100644
--- a/net/url_request/url_request_context_getter.h
+++ b/net/url_request/url_request_context_getter.h
@@ -79,11 +79,16 @@ class NET_EXPORT URLRequestContextGetter
   friend class web::NetworkContextOwner;
 #endif  // defined(OS_IOS)
 
+  // NOTE(nornagon/robo): electron currently depends on this deprecated API
+  // from net/, but will be removing it soon when we switch to using the
+  // network service.
+ public:
   // Adds / removes an observer to watch for shutdown of |this|'s context. Must
   // only be called on network thread. May not be called once
   // GetURLRequestContext() starts returning nullptr.
   void AddObserver(URLRequestContextGetterObserver* observer);
   void RemoveObserver(URLRequestContextGetterObserver* observer);
+ private:
 
   // OnDestruct is used to ensure deletion on the thread on which the request
   // IO happens.
